schema/stacks/stack_schema_merge.go

```diff
@@ -196,5 +215,111 @@ func schemaForDependentComponentBlock(modMeta *tfmod.Meta) (*schema.BodySchema,
 		},
 	}
 
+	if component.Source == "" {
+		// avoid creating output refs if we don't have reference name
+		return bodySchema, nil
+	}
+
+	modOutputTypes := make(map[string]cty.Type, 0)
+	modOutputVals := make(map[string]cty.Value, 0)
+	targetableOutputs := make(schema.Targetables, 0)
+	impliedOrigins := make(schema.ImpliedOrigins, 0)
+
+	for name, output := range modMeta.Outputs {
+		// TODO: figure out how to handle this better
+		component.Source = strings.TrimPrefix(component.Source, "./")
+		addr := lang.Address{
+			lang.RootStep{Name: "component"},
+			lang.AttrStep{Name: component.Source},
```

Comment:
using the source here only works if the component name (label 1) matches the source – I'm going to add a commit that will pass down the component name to `schemaForDependentComponentBlock` and change this step here and further down below.

Reply:
Nice work!